### PR TITLE
QIE10 bug fixes

### PR DESCRIPTION
**Pull request by kpedro88**
This PR fixes a few QIE10-related bugs for HF. It will be backported to 80X.

### CODE REVIEW

**Comment by mmusich** [comment]
please test

**Comment by kpedro88** [comment]
@mmusich @ggovi - the comparison looks fine to me. QIE10s currently aren't used in any central workflows. Can you sign this one and the backport #13833?

**Comment by kpedro88** [comment]
@mmusich @ggovi - if even simple bug fix PRs take a week to get approved, the HCAL Phase 1 software will never converge...

**Comment by mmusich** [comment]
@kpedro88 sorry back only now from Easter break. We'll have a look.

**Comment by kpedro88** [comment]
@mmusich @ggovi - 10 days and counting for a PR with 9 changed lines...

**Comment by mmusich** [comment]
@kpedro88 2016 comes before 2017

**Comment by mmusich** [comment]
@ggovi this is mostly for you, please give a look

**Comment by mmusich** [comment]
+1

**Comment by ggovi** [comment]
+1

File: CalibFormats/HcalObjects/src/HcalCoderDb.cc [hunk 1]
```diff
@@ -23,9 +23,12 @@ template <class Digi> void HcalCoderDb::adc2fC_ (const Digi& df, CaloSamples& cl
 
 template <> void HcalCoderDb::adc2fC_<QIE10DataFrame> (const QIE10DataFrame& df, CaloSamples& clf) const {
   clf=CaloSamples(df.id(),df.samples());
+  int presample = -1;
   for (int i=0; i<df.samples(); i++) {
     clf[i]=mCoder->charge (*mShape, df[i].adc (), df[i].capid ());
+    if(df[i].soi()) presample = i;
```
Comment (davidlange6) [inline CalibFormats/HcalObjects/src/HcalCoderDb.cc]: @kpedro88 - this is the same as 

clf.setPresamples(i);

or not (saving the temp variable and second if)?

**Comment by mmusich** [comment]
please test

**Comment by kpedro88** [comment]
@davidlange fixed

**Comment by ggovi** [comment]
+1

**Pull request merged**